scripts/bfasst/compare_waveforms/file_parsing/parse_files.py

```diff
@@ -1,8 +1,29 @@
+"""Handles parsing the Verilog files to find the signal names and lengths."""
 import spydrnet as sdn
 
 
+def get_data(data, ports):
+    """Finds all of the IO ports in the netlist and stores them in data."""
+    for port in ports:
+        if str(port.direction) == "Direction.OUT":
+            data["output_list"].append(port.name)
+            data["total_list"].append(port.name)
+            data["output_bits_list"].append(len(port.pins) - 1)
+        elif str(port.direction) == "Direction.IN":
+            data["input_list"].append(port.name)
+            data["total_list"].append(port.name)
+            data["input_bits_list"].append(len(port.pins) - 1)
+    if "clk" in data["input_list"]:
+        if data["input_list"].index("clk") == 0:
+            data["input_list"].append(data["input_list"].pop(0))
+    return data
+
+
 def parse(file):
-    """Uses spydrnet to analyze the netlist and add the names of all inputs, outputs, and their respective bit sizes to the data structure."""
+
+    """Uses spydrnet to analyze the netlist and add the names of all inputs, outputs, and their
+    respective bit sizes to the data structure."""
+
     netlist = sdn.parse(str(file))
     library = netlist.libraries[0]
     definition = library.definitions[0]
```

Comment:
Consider using `netlist.top_instance().parent()` to get the definition.  This has the added benefit of working with edif files.

Reply:
I'm running some tests on this function and currently it's telling me that parent definition is undefined. Reference definition seems to be defined, but I'm not sure what that really does. I'll see if the documentation can clear that up.